### ISSUE TITLE
Crash if nix-diff tries to open a derivation that doesn't exist

### ISSUE THREAD

**Issue by ocharles** [Gabriel439/nix-diff#3]
I have `scp`ed just a single derivation from a remote machine. If I try and diff this against a local diff, `nix-diff` is productive, but if the input derivations change, `nix-diff` seems to try and recurse into them - but that fails as I didn't copy the input derivations. It would be better if `nix-diff` could handle this and warn me that it couldn't continue doing a deep comparison.

**Comment by Gabriel439** [comment]
Yeah, this seems reasonable.  Would you be able to contribute a fix for this one?
